src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -150,58 +155,25 @@ private void handleMemberRemovedEvent(final String message) {
         }
     }
 
-    @SuppressWarnings("rawtypes")
-    private static String extractDataStringFrom(final String message) {
-        final Map jsonObject = GSON.fromJson(message, Map.class);
-        return  (String) jsonObject.get("data");
-    }
+    public String extractUserIdFromChannelData(final String channelDataString) {
+        try {
+            ChannelData data = GSON.fromJson(channelDataString, ChannelData.class);
 
-    @SuppressWarnings("rawtypes")
-    private static PresenceData extractPresenceDataFrom(final String message) {
-        final String dataString = extractDataStringFrom(message);
-        return GSON.fromJson(dataString, Presence.class).presence;
-    }
+            if (data.getUserId() == null) {
```

Comment:
Is there a way to get GSON to fail if the field doesn't exist - rather than having to check for null here?